onnxscript/function_libs/torch_lib/ops/nn.py

```diff
@@ -2069,9 +2069,9 @@ def _aten_scaled_dot_product_attention_bool_mask_onnx(
     query_scaled = op.Mul(query, op.Sqrt(scale))
     key_transposed_scaled = op.Mul(key_transposed, op.Sqrt(scale))
     # Turn the Boolean mask to float: attn_mask.masked_fill(not attn_mask, -float('inf'))
-    attn_mask = op.Where(
-        attn_mask, op.Constant(value_float=0.0), op.Constant(value_float=-float("inf"))
-    )
+    zero = op.CastLike(op.Constant(value_float=0.0), query_scaled)
+    neg_inf = op.CastLike(op.Constant(value_float=-float("inf")), query_scaled)
+    attn_mask = op.Where(attn_mask, zero, neg_inf)
```

Comment:
Can we create the tensor directly using query.dtype? We can use `op.Constant(value=ir.tensor(...))`